File: BubbleSort/Ruby/BubbleSort.rb
def bubble_sort(list)
  return list if list.size <= 1 
  swapped = true
  while swapped do
    swapped = false
    0.upto(list.size-2) do |i|
      if list[i] > list[i+1]
        list[i], list[i+1] = list[i+1], list[i] 
        swapped = true
      end
    end    
  end

  list
end

File: Rust/Bubble Sort/BubbleSort.rs
/*
 * Implementation of BubbleSort in Rust
 */

fn bubble_sort(mut list: Vec<i32>) -> Vec<i32> {
    let mut swapped = true;
    while swapped {
        swapped = false;
        for i in 0..list.len()-1 {
            if list[i] > list[i+1] {
                // Swap
                let s = list.remove(i);
                list.insert(i+1, s);
                swapped = true;
            }
        }
    }

    return list;
}
